src/hip_module.cpp

```diff
@@ -144,137 +144,121 @@ hipError_t ihipModuleLaunchKernel(TlsData *tls, hipFunction_t f, uint32_t gridSi
     using namespace hip_impl;
 
     auto ctx = ihipGetTlsDefaultCtx();
-    hipError_t ret = hipSuccess;
 
-    size_t globalWorkSizeX = (size_t)gridSizeX * (size_t)blockSizeX;
-    size_t globalWorkSizeY = (size_t)gridSizeY * (size_t)blockSizeY;
-    size_t globalWorkSizeZ = (size_t)gridSizeZ * (size_t)blockSizeZ;
-    if(globalWorkSizeX > UINT32_MAX || globalWorkSizeY > UINT32_MAX || globalWorkSizeZ > UINT32_MAX)
-    {
+    if (!ctx) return hipErrorInvalidDevice;
+
+    if (gridSizeX > UINT32_MAX / blockSizeX ||
+        gridSizeY > UINT32_MAX / blockSizeY ||
+        gridSizeZ > UINT32_MAX / blockSizeZ) {
         return hipErrorInvalidConfiguration;
     }
 
-    if (ctx == nullptr) {
-        ret = hipErrorInvalidDevice;
-
-    } else {
-        int deviceId = ctx->getDevice()->_deviceId;
-        ihipDevice_t* currentDevice = ihipGetDevice(deviceId);
-        hsa_agent_t gpuAgent = (hsa_agent_t)currentDevice->_hsaAgent;
-
-        std::vector<char> kernargs{};
-        if (kernelParams) {
-            if (extra) return hipErrorInvalidValue;
+    int deviceId = ctx->getDevice()->_deviceId;
+    ihipDevice_t* currentDevice = ihipGetDevice(deviceId);
+    hsa_agent_t gpuAgent = (hsa_agent_t)currentDevice->_hsaAgent;
 
-            for (auto&& x : f->_kernarg_layout) {
-                const auto p{static_cast<const char*>(*kernelParams)};
+    std::vector<char> kernargs{};
+    if (kernelParams && *kernelParams) {
+        if (extra) return hipErrorInvalidValue;
 
-                kernargs.insert(
-                    kernargs.cend(),
-                    round_up_to_next_multiple_nonnegative(
-                        kernargs.size(), x.second) - kernargs.size(),
-                    '\0');
-                kernargs.insert(kernargs.cend(), p, p + x.first);
+        for (auto&& x : f->_kernarg_layout) {
+            const auto p{static_cast<const char*>(*kernelParams)};
 
-                ++kernelParams;
-            }
-        } else if (extra) {
-            if (extra[0] == HIP_LAUNCH_PARAM_BUFFER_POINTER &&
-                extra[2] == HIP_LAUNCH_PARAM_BUFFER_SIZE && extra[4] == HIP_LAUNCH_PARAM_END) {
-                auto args = (char*)extra[1];
-                size_t argSize = *(size_t*)(extra[3]);
-                kernargs.insert(kernargs.end(), args, args+argSize);
-            } else {
-                return hipErrorNotInitialized;
-            }
+            kernargs.insert(
+                kernargs.cend(),
+                round_up_to_next_multiple_nonnegative(
+                    kernargs.size(), x.second) - kernargs.size(),
+                '\0');
+            kernargs.insert(kernargs.cend(), p, p + x.first);
 
-        } 
-        else if (f->_kernarg_layout.size() != 0) {
```

Comment:
I think last merge of master into this PR branch lost this check.  That's why the test `directed_tests/gcc/LaunchKernel.tst` is now reporting OTHER_FAULT in the CI.

If I add this case back in, I end up with a seg fault instead, because `extra` is null and is unconditionally dereferenced later.